.github/workflows/apps/latest_major_version.go

```diff
@@ -0,0 +1,153 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"sort"
+	"github.com/Masterminds/semver/v3"
+	"golang.org/x/mod/modfile"
+	"net/http"
+	"os"
+	"regexp"
+	"strings"
+)
+
+type Tag struct {
+	Name string
+}
+
+func getLatestMajorVersion(repo string) (string, error) {
+	// Get latest major version available for repo from github.
+	const apiURL = "https://api.github.com/repos/%s/tags"
+	url := fmt.Sprintf(apiURL, repo)
+
+	resp, err := http.Get(url)
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return "", fmt.Errorf("failed to fetch tags: %s", resp.Status)
+	}
+
+	var tags []struct {
+		Name string `json:"name"`
+	}
+
+	if err := json.NewDecoder(resp.Body).Decode(&tags); err != nil {
+		return "", err
+	}
+	latestByMajor := make(map[int]*semver.Version)
+
+	for _, tag := range tags {
+		v, err := semver.NewVersion(tag.Name)
+		if err != nil {
+			continue // Skip invalid versions
+		}
+
+		if v.Prerelease() != "" {
+			continue // Ignore pre-release versions
+		}
+
+		major := int(v.Major())
+		if current, exists := latestByMajor[major]; !exists || v.GreaterThan(current) {
+			latestByMajor[major] = v
+		}
+	}
+
+	var latestMajor *semver.Version
+	for _, v := range latestByMajor {
+		if latestMajor == nil || v.Major() > latestMajor.Major() {
+			latestMajor = v
+		}
+	}
+
+	if latestMajor != nil {
+		return fmt.Sprintf("v%d", latestMajor.Major()), nil
+	}
+
+	return "", fmt.Errorf("no valid versions found")
+
+}
+
+func main() {
+
+	data, err := os.ReadFile("integration_go.mod")
```

Comment:
@quinna-h What's the purpose of this `integration_go.mod` file?

Reply:
We were originally parsing the `go.mod`, but there was no straightforward way to read _only_ the packages that are integrations in the `go.mod`. So we (me, @bouwkast , and @rarguelloF  ) decided it would be best to have a 'dummy' `integration_go.mod` file, as a workaround

Reply:
I guess it won't cause any issue, so I'm approving. Although, I'd be maybe more comfortable if it's stored in a different place than the root, but I see no other objective reason for moving it. So, let's leave it in the root.